### ISSUE TITLE
ROADMAP — Ruse Native generations: educate, preserve, promote, specialise

### ISSUE THREAD

**Issue by bluegrouse92-lgtm** [N-A-pixel/Ruse-Public#6]
# Ruse Native — possible generational roadmap

This issue records a **possible long-term experimental roadmap**, not a promise that RN-20M or any successor will reach a predetermined capability level.

The roadmap follows the evidence gathered so far: preserve each generation, educate it properly, measure its real ceiling, and use what it teaches us to design the next generation. Older useful generations do not need to be discarded when a stronger native model arrives; they may become specialised experimental workers if evaluation shows they are genuinely good at a bounded task.

## Starting point — RN-20M, Generation 1

```text
Model:       RN-20M
Family:      Ruse Native
Class:       20M
Historical early count: 21,880,704
Authoritative OG artifact count: 21,865,344
Seed:        20260921
Generation:  1
```

Current evidence:

- Stage 0 / BIRTH: PASS — model lifecycle and native runtime proven.
- Stage 1: PASS — RN-20M measurably learns.
- Stage B2: FAIL — much better emitted answers, but 0/6 held-out compositions.
- Stage B3: FAIL — atomic components removed the spelling/tokenization excuse, but held-out composition remained 0/12.
- Foundation C0: feasibility complete — real-language causal pretraining works on the Steam Deck at roughly 652 tokens/sec inside the measured resource envelope.
- Tokenizer V1 required whitespace/newline correction before scale.

The lesson from B2/B3 is not simply "train longer." RN-20M needs a real language foundation before increasingly difficult generalisation tests become meaningful.

# Phase 1 — educate RN-20M properly

Immediate order of work:

```text
Tokenizer V1 diagnosis
→ Tokenizer V1.1 if required
→ exact whitespace/newline round-trip verification
→ resumable / incremental trainer
→ cheap checkpoint-resume dry run
→ bounded foundation-training increments
```

Possible observation points include ~100K, ~1M, ~10M, ~50M and ~100M+ tokens. These are observation scales, not guaranteed targets.

At each meaningful checkpoint preserve:

- model weights and hash
- tokenizer and hash
- exact corpus/data manifest
- optimizer/training state needed for resume
- train and validation loss
- fixed raw generation probes
- held-out evaluations
- resource/throughput measurements
- checkpoint save/reload proof
- known failures and regressions

The objective is a developmental curve rather than one final checkpoint.

# Phase 2 — find the Generation-1 ceiling

Continue only while evidence shows useful improvement. Eventually look for a real plateau across language coherence, instruction following, held-out generalisation, compositional behaviour, structured-output reliability, latency and resource cost.

A capacity ceiling is **not a failed lineage**. It is evidence about what the architecture can and cannot do.

# Phase 3 — freeze RN-20M and discover useful bounded roles

When a stronger native generation exists, freeze the best RN-20M general checkpoint first. Then evaluate it across a worker-qualification suite to discover where it is genuinely strong.

Possible experimental roles include:

- intent classification/routing
- structured extraction
- command normalization
- safety/preflight checks
- cheap first-pass classification
- narrow protocol tasks

These are candidate roles only. A role is earned only by benchmark evidence.

If a useful specialty emerges, create a separately versioned specialist descendant rather than overwriting the frozen general checkpoint.

# Phase 4 — Generation 2

If Generation 1 exposes a genuine capacity or architecture ceiling, design Generation 2 from the evidence rather than merely rerolling RN-20M's seed.

Generation 2 should begin with a predeclared design informed by RN-20M:

- parameter budget justified by measured hardware capacity
- tokenizer lessons carried forward
- architecture weaknesses addressed where evidence supports doing so
- improved corpus/data mixture
- improved training schedule
- clean new seed
- clean Day Zero
- separate immutable lineage

# Phase 5 — persistent experimental model family

The longer-term research direction is a family of preserved model lineages whose useful members can be assigned evidence-backed bounded roles.

```text
RUSE NATIVE EXPERIMENT
        │
   capability registry
        │
 ┌──────┼──────┐
 ▼      ▼      ▼
RN-20M  RN-G2  RN-G3
small   larger specialist
worker  worker worker
```

Workers need not remain resident continuously; they can be loaded only when a benchmark-qualified role requires them.

# Phase 6 — capability registry

A future capability record can include:

```text
worker_id
lineage
parent_checkpoint
specialty
qualification_suite
qualification_score
known_failure_domains
context_limit
latency_profile
memory_cost
last_requalified
```

Routing decisions should be based on evidence, not parameter count or model age.

# Phase 7 — specialisation without catastrophic forgetting

Preserve both the best general checkpoint and separately versioned specialist descendants:

```text
best general Generation-N checkpoint   ← immutable
                 │
                 ├── specialist A descendant
                 ├── specialist B descendant
                 └── experimental descendants
```

# Phase 8 — successors inherit access, not necessarily weights

A later model does not need every useful ancestral skill copied into its own parameters. Experimental system-level capability can accumulate through access to qualified specialists while individual models retain finite capacity.

# Phase 9 — external models remain comparison/escape-hatch tools

External/open models, APIs or specialist systems may still be used for comparison, teaching or bounded work where a native model does not yet have the required capability. They remain replaceable dependencies of the experiment.

# Phase 10 — preserve the family tree

Every native generation should have a reproducible birth record containing:

```text
model designation
architecture
exact parameter count
seed
generation number
tokenizer lineage
Day Zero model hash
training-data manifests
major checkpoint hashes
predeclared evaluations
promotion decision
qualified role(s), if any
specialist descendant hashes
```

Failed generations and failed experiments remain part of the record.

## Roadmap update — RN-8M quadruplets

The roadmap now includes a concrete four-model RN-8M cohort: **RN-8M-Q1/Q2/Q3/Q4**, each 8,523,008 parameters.

They retain separate neural and optimizer state while receiving synchronized education and bounded frozen peer-output exposure at controlled barriers.

Their broad path is:

```text
shared foundation classroom
→ freeze four general checkpoints
→ identical no-update specialist-discovery evaluation
→ preserve behavioural/developmental fingerprints
→ identify evidence-supported candidate roles
→ separately authorize specialist descendants where justified
```

No specialist role is assigned during common education. Useful divergence is preserved rather than normalized away.

The governing principle remains:

> **Preserve the stages. Preserve the lineage. Measure first. Specialise only where evidence supports it.**

**Comment by bluegrouse92-lgtm** [comment]
## Roadmap update — E0 family exists

Controlled E0 completed for RN-20M Clone, corrected RN-50M and RN-100M on the same frozen 258,816-token curriculum. All three completed safely and passed checkpoint/reload verification.

The preserved 49.75M-parameter 32K model also remains a separate multilingual branch rather than being discarded.

The roadmap principle is now concrete: preserve lineage first, then qualify useful branches from measured evidence.

**Comment by bluegrouse92-lgtm** [comment]
## Roadmap update after E3

RN-20M Clone, corrected RN-50M and RN-100M reached the common E3 boundary at 5,000,064 tokens. Gary's separate 32K lineage reached the same developmental rung.

Next steps: preserve E3 checkpoints, resume OG RN-20M, expand with the RN-8M cohort, a second ~100M model and RN-300M, then compare capabilities under controlled evaluation.

**Comment by bluegrouse92-lgtm** [comment]
## E3-EI — experimental descendant frozen

A small targeted English intervention produced strong behavioral redistribution across RN-20M-clone, RN-50M-corrected, RN-100M and Gary, but did not reliably fix topic drift, weak factual completion, weak causal continuity or repetition.

Historical validation/holdout loss worsened for all four. The result is preserved as useful curriculum evidence, but the E3-EI branch is **not promoted**. Canonical E3 checkpoints remain the main-line historical boundary.

**Comment by bluegrouse92-lgtm** [comment]
## 23 September 2026 — original Ruse Native tree resumes growth

The protected Generation-1 record remains immutable, but new births and controlled descendants may now be created alongside it.

The active population now includes OG RN-20M, RN-8M-Q1–Q4, a fresh RN-100M and RN-300M-new-20260923, in addition to the already-educated E3 family and Gary's separate 32K multilingual lineage.

These models exist to create controlled differences in scale, initialization, tokenizer lineage and developmental history. Historical checkpoints remain preserved rather than overwritten.

**Comment by bluegrouse92-lgtm** [comment]
## 23 September 2026 — Ruse School Phase 2 complete

RN-300M's first end-to-end pre-education pipeline on Oracle is operational. The package/admission boundary, canonical ownership transfer, proof-library admission, deterministic curriculum construction, tokenizer verification, novelty/contamination checks and training lock were all exercised without mutating the model.

The proof shelf admitted five public-domain documents; one duplicate and one rejected probe were excluded. The first frozen lesson contained 3,991 tokens and passed tokenizer, novelty, contamination and provenance checks.

The training lock correctly refused an unauthorized launch. At this milestone RN-300M remained at zero training tokens and the frozen lesson had not been applied.

**Comment by bluegrouse92-lgtm** [comment]
## 24 September 2026 — RN-8M +5M classroom launched

The quadruplet programme advanced from the completed 400K classroom into a new synchronized common-education stage.

Each sibling starts at lifetime age **400,126** and is authorized for exactly **+5,000,000 optimized targets**, ending at **5,400,126** if completed canonically.

The stage uses five synchronized +1M barriers. Each interval contains **998,976 common-base targets + 1,024 sibling-review targets**, with peer review built only from the other three siblings and only from already-frozen prior-barrier outputs.

The full stage accounts for **39,065 optimizer updates / 5,000,000 loss-contributing targets per sibling**. Single-writer protection and explicit crash recovery were qualified before launch.

The execution ledger records `training_started`; no specialization assignment is made during this common-education stage.
